### DISCUSSION TITLE
Simulating ~7000 agricultural fields with APSIMx?

### DISCUSSION THREAD

**Discussion by i-heis** [APSIMInitiative/ApsimX#8743] [Q&A] [answered]
Dear community,
I am looking for an agricultural model to simulate management impacts on ecosystem services such as water quality, drought protection and carbon sequestration (and yields of course) in a study region in Germany. APSIMx is currently the favorite. We stumbled upon one thing: the faktorial implementation of simulation fields/cells in the gui. **My study region comprises ~7000 agricultural fields and we were wondering if somebody has experience with simulating so many fields with in principle different properties?** Our first thought was, that it is possible to summarize main types of fields regarding biophysical properties and management to reduce the number of fields that one needs to set up (manually) in the gui. 
Has someone already used a work around where setting up fields was automatised with e.g. a python script or something similar? Because even if we would summarise the fields into fewer property-groups, a manual set up seems prone to typing mistakes.
Thanks for your replies & cheers
Irina

**Comment by LenLon** [comment, marked as answer]
Hello Irina,

I am a PhD student working on similar things and managed to setup APSIM in a way that makes running these large-scale simulations not easy, but at least feasible. I work with the PIK and we are currently developing a small suite of scripts to generate weather and soil data for these large-scale sims.

Using APSIM-X to work on larger scales beyond a single field is not a common use case if you look at the documentation, but is certainly done though the scale and applicability depends a lot on your infrastructure (i.e. computer resources) and goals. A factorial implementation seems to not be what you want, as I assume the 7,000 fields are not part of a controlled environment where only nitrogen fertilisation or the cultivar planted is the differing factor, and you might have different weather and soils between them as well. 

What is commonly done for these larger studies (e.g. in global gridded crop modelling) is that indeed APSIM-x is either called via a bash script using command line or using R, with soils and weather data as separate input files for each simulation site (or all soils already present in the simulation). I cannot go into detail here but a starting point is the apsimx-R package by @femiguez . Implementation depends on your goals (do you want to parallelise? how many different treatments and years are you planning to model per field?), but it usually involves writing an R script that calls APSIM-X in a loop, replacing soils and weather data and management variables according to a pre-defined list, runs the simulation in APSIM and then saves the outputs to another file. 
If you want to run APSIM from the command line, there are a number of issues/threads here on the Github where the developers give some advice on how to achieve that as well.

If you have more questions you can also write me an e-mail: lennart.jansen@uni-kassel.de

Best
Lennart

**Reply by i-heis** [reply]
Thanks for your detailed answer, it was very helpful to understand a few more things. I will start getting more accustomed to APSIMx and then come back to you at some point :) Cheers, Irina

**Comment by HamishBrownPFR** [comment]
@ed2014 and @rcichota have plenty of experience with such large simulations using Next Generation.  Normally it is not done using the experiment functionality as the simulation job is too big for a single computer.   Instead, it is done on HPC using a base simulation and libraries of soils and climates and scripts to stick the correct soil and climate into the base simulation to represent each specific paddock.

**Reply by i-heis** [reply]
Thanks for your reply Hamish, very useful to get a better idea of the model. We can use an HPC environment, so this wouldn't be a problem. I am not very familiar with APSIMx yet, but the experiment functionality is basically the function where you can automatically vary parameters for your model runs right? First of all, we would work with a 'status quo' scenario where site parameters are constant and only management variables would be varied. Later on, we are planning to include 3 climate scenarios and test out different management scenarios/bundles.

**Reply by BrianCollinss** [reply]
Hey @i-heis . In what capacity do you need the help? Is this a PhD or postdoc project? Do you need someone to help you set things up, or someone to run the sims for you?

**Reply by i-heis** [reply]
Hey @BrianCollinss, thanks for your message. I am using APSIM-x for my PhD project which will last another 3 years. I am in touch with Iris Vogeler at the University Aarhus in Denmark, who has some experience with APSIM classic and German modelling studies. However, I am always grateful for support with (future) set up issues. I am still in the very early phase of getting to know APSIM, but I am open to connect with you or others and give you an idea of my modelling objectives. You can just write me an email: irina.heiss@ufz.de
Thanks :)

**Reply by BrianCollinss** [reply]
Thanks Irina. Our team at UniSQ welcomes any collaboration, especially in a formal format. If you think you are going to need assistance or even a co-supervisor, please reach out to me at brian.collins@unisq.edu.au. Ta
